ecl/hql/hqlfold.cpp

```diff
@@ -481,6 +492,17 @@ static IHqlExpression * optimizeCompare(IHqlExpression * expr)
     else if (rightValue)
     {
         ITypeInfo * lType = leftChild->queryType();
+        bool isUnsigned = lType->getTypeCode()==type_int && !lType->isSigned();
```

Comment:
Could also match decimal, swapint, and packed int.  Worth extracting into an isUnsignedType()